packages/apps/dashboard/ui-2024/src/features/Leaderboard/hooks/useDatagridOptions.tsx

```diff
@@ -0,0 +1,132 @@
+import { Typography } from '@mui/material';
+import {
+  GridColDef,
+  GridRenderCellParams,
+  GridSortModel,
+  GridCallbackDetails,
+  GridValidRowModel,
+} from '@mui/x-data-grid';
+import { GridApiCommunity } from '@mui/x-data-grid/models/api/gridApiCommunity';
+import { useMemo, useCallback, MutableRefObject } from 'react';
+import { useBreakPoints } from '@utils/hooks/use-is-mobile';
+
+interface IColPossibleOptions {
+  isRowIdx?: boolean;
+  pinnedColumnName?: string;
+}
+
+interface IProps<T extends GridValidRowModel> extends IColPossibleOptions {
+  columns: GridColDef<T>[];
+  apiRef: MutableRefObject<GridApiCommunity>;
+}
+
+export function useDatagridOptions<T extends GridValidRowModel>({
+  apiRef,
+  columns,
+  isRowIdx,
+  pinnedColumnName,
+}: IProps<T>) {
+  const {
+    mobile: { isMobile },
+  } = useBreakPoints();
+
+  const handlePinnedColOption = useCallback(
+    (cols: GridColDef<T>[]) =>
+      cols.map((col) =>
+        col.field === pinnedColumnName
+          ? {
+              ...col,
+              headerClassName: isMobile
+                ? 'home-page-table-header pinned-column--header'
+                : 'home-page-table-header',
+              cellClassName: isMobile ? 'pinned-column--cell' : '',
+            }
+          : col
+      ),
+    [isMobile, pinnedColumnName]
+  );
+
+  const handleRowOrderNum = useCallback(
+    (cols: GridColDef<T>[]) => {
+      return isMobile
+        ? cols
+        : [
+            {
+              field: 'id',
```

Comment:
```suggestion
              field: 'rowIndex',
```

and `{params.value + 1}` below

Reply:
Good catch again, sorry for oversight :) fixed